instill/clients/artifact.py

```diff
@@ -0,0 +1,442 @@
+# pylint: disable=no-member,wrong-import-position,too-many-lines,no-name-in-module
+from typing import Dict
+
+# pipeline
+import instill.protogen.artifact.artifact.v1alpha.artifact_pb2 as artifact_interface
+import instill.protogen.artifact.artifact.v1alpha.artifact_public_service_pb2_grpc as artifact_service
+import instill.protogen.artifact.artifact.v1alpha.chunk_pb2 as chunk_interface
+
+# common
+import instill.protogen.common.healthcheck.v1beta.healthcheck_pb2 as healthcheck
+from instill.clients.base import Client, RequestFactory
+from instill.clients.constant import DEFAULT_INSTANCE
+from instill.clients.instance import InstillInstance
+from instill.configuration import global_config
+from instill.utils.error_handler import grpc_handler
+
+
+class ArtifactClient(Client):
+    def __init__(self, async_enabled: bool) -> None:
+        self.hosts: Dict[str, InstillInstance] = {}
```

Comment:
not sure if we need to set a default namespace like other clients, its your call @joremysh

Reply:
It seems the `namespace_id` format is like `jeremyshih` which is different from the `target_namespace` in pipeline (`users/jeremyshih`).
So I assume we could keep it for now and set it when needed